x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/InferencePyTorchAction.java

```diff
@@ -109,6 +111,17 @@ protected void doRun() throws Exception {
         } catch (IOException e) {
             logger.error(() -> "[" + getDeploymentId() + "] error writing to inference process", e);
             onFailure(ExceptionsHelper.serverError("Error writing to inference process", e));
+        } catch (ElasticsearchException e) {
+            // Don't log problems related to the shape of the input as errors
+            if (e.status().getStatus() >= RestStatus.INTERNAL_SERVER_ERROR.getStatus()) {
+                logger.error(() -> "[" + getDeploymentId() + "] internal server error running inference", e);
```

Comment:
Since we are logging this error for all rest statuses >=500, perhaps it would be best to remove the word "internal" from the logging message, as "Internal Server Error" is only the 500 code itself.

Reply:
The other 5xx error codes still relate to things that are internal to the server, just more specific than 500.

These error logs are really for our benefit rather than the end user's, and one important thing about them is that we can easily find which line of code logged them when we see them in the logs. When someone sees one of these in a log file their next question will be, "What could have caused that?" rather than, "Are the exact words used in the message pedantically correct?"